src/components/settingsdialog/SettingsSelectors.tsx

```diff
@@ -0,0 +1,257 @@
+import { FormControl, FormHelperText, MenuItem, Select, SelectChangeEvent } from '@mui/material';
+import { updateMic, updatePreviewMic, updatePreviewWebcam, updateScreenSharing, updateWebcam } from '../../store/actions/mediaActions';
+import { useAppDispatch, useAppSelector } from '../../store/hooks';
+import { settingsActions } from '../../store/slices/settingsSlice';
+import { Resolution } from '../../utils/types';
+import {
+	selectAudioChannelCountLabel,
+	selectAudioSampleRateLabel,
+	selectAudioSampleSizeLabel,
+	selectOpusPtimeLabel,
+	selectRecordingsPreferredMimeTypeLabel,
+	selectResolutionLabel,
+	selectScreenSharingFrameRateLabel,
+	selectWebcamFrameRateLabel
+} from '../translated/translatedComponents';
+
+interface ResolutionSelectorProps {
+	resolutions: Array<{ value: Resolution, label: () => string }>
+}
+
+export const ResolutionSelector = ({
+	resolutions
+}: ResolutionSelectorProps): JSX.Element => {
+	const dispatch = useAppDispatch();
+	const resolution = useAppSelector((state) => state.settings.resolution);
+
+	const handleResolutionChange = (event: SelectChangeEvent<string>): void => {
+		dispatch(updateWebcam({
+			restart: true,
+			newResolution: event.target.value
+		}));
+		dispatch(updatePreviewWebcam({
```

Comment:
Let's not do both of these at every change as that is very heavy. Changing the main stream is enough, so remove the preview stream update.